File: sorting/selection_sort/selection_sort.go
package selection_sort

func selectionSort(nums []int) {
	n := len(nums)
	for i := 0; i < n-1; i++ {
		minIdx := i

		for j := i + 1; j < n; j++ {
			if nums[j] < nums[minIdx] {
				minIdx = j
			}
		}

		nums[i], nums[minIdx] = nums[minIdx], nums[i]
	}
}

File: sorting/selection_sort/selection_sort.cpp
#include <iostream>
#include <vector>

void selectionSort(std::vector<int>& nums) {
  size_t n = nums.size();

  for (size_t i = 0; i < n-1; ++i) {
    int minIdx = i;

    for (size_t j = i+1; j < n; ++j) {
      if (nums[j] < nums[minIdx]) minIdx = j;
    }

    std::swap(nums[i], nums[minIdx]);
  }
}